include/deal.II/base/subscriptor.h

```diff
@@ -92,6 +92,15 @@ class Subscriptor
    */
   Subscriptor &operator = (const Subscriptor &);
 
+#ifdef DEAL_II_WITH_CXX11
+  /**
+   * Move assignment operator.
+   *
+   * Asserts that the counter for the moved object is zero.
+   */
+  Subscriptor &operator = (Subscriptor &&);
```

Comment:
I was a bit misleading - I'm happy with this implementation. It is far
more fledged out and thought through than my initial attempt on
providing move operators.